src/update_handler.c

```diff
@@ -952,6 +952,17 @@ static gboolean ubifs_format_slot(RaucSlot *dest_slot, GError **error)
 
 	g_ptr_array_add(args, g_strdup("mkfs.ubifs"));
 	g_ptr_array_add(args, g_strdup("-y"));
+
+	if (dest_slot->extra_mkfs_opts != NULL) {
+		gchar **extraopts = g_strsplit(dest_slot->extra_mkfs_opts, " ", 0);
```

Comment:
This code is duplicated for each filesystem. I'd prefer to handle this during the config file parsing. 
Also, use `g_shell_parse_argv()`, similar to `casync_install_args` in `casync_extract()`. That will also make error detection easier.

Each format function would then only use our `r_ptr_array_addv()` helper.